app/packages/looker/src/worker/canvas-decoder.ts

```diff
@@ -121,6 +128,31 @@ export const decodeWithCanvas = async (blob: Blob, cls: string) => {
     channels = 1;
   }
 
+  // if it's segmentation, we need to recast according to whether or not this field is mapped to RGB targets
+  if (cls === SEGMENTATION) {
+    let maskTargets = coloring.maskTargets?.[field];
+    if (maskTargets === undefined) {
+      maskTargets = coloring.defaultMaskTargets;
+    }
+    const isRgbMaskTargets_ = isRgbMaskTargets(maskTargets);
+
+    if (!isRgbMaskTargets_ && channels > 1) {
+      // recast to mono channel because we don't need the other channels
+      targetsBuffer = recastBufferToMonoChannel(
+        imageData.data,
+        width,
+        height,
+        channels
+      );
+      channels = 1;
+    }
+
+    // note: for JPG segmentations with RGB mask targets, we don't need to recast
+    // although depending on the JPG compression, we might have some artifacts.
+    // even the slightest change in color can cause the mask to be rendered as
+    // background color (transparent) instead of the actual mask color
```

Comment:
I wouldn't be opposed to considering this a hard error if it is easy to present/document

Reply:
Yeah... the thing is you can have JPGs with zero compression and that might not be a problem, so probably not the best idea to consider it a hard error.